src/lib/loaders/loaders_without_authentication/search.ts

```diff
@@ -0,0 +1,23 @@
+// @ts-check
+import factories from "../api"
+import * as url from "url"
+import { SearchIndex } from "schema/search"
+
+export const searchLoader = opts => {
+  const { gravityLoaderWithoutAuthenticationFactory } = factories(opts)
+  const gravityLoader = gravityLoaderWithoutAuthenticationFactory
+
+  return gravityLoader(({ query, indexes, mode }) => {
+    const queryParams = {
+      term: query,
+      "indexes[]": indexes || SearchIndex.getValues().map(index => index.value),
+    }
+
+    switch (mode) {
+      case "AUTOSUGGEST":
+        return url.format({ pathname: "/match/suggest", query: queryParams })
+      default:
+        return url.format({ pathname: "/match", query: queryParams })
+    }
+  })
+}
```

Comment:
I might be missing something, but it seems like this is identical to the authenticated one and so making the change to inject the already created `gravityLoader` would remove the need to have two copies of the `searchLoader`.

Also, we're following a file naming convention where the file is called the same as the main export, i.e. `searchLoader.ts`.